pvnet/app.py

```diff
@@ -117,23 +118,42 @@ def convert_df_to_forecasts(
 
     forecasts = []
 
-    for gsp_id in forecast_values_df.gsp_id.unique():
+    for gsp_id in forecast_values_dataarray.gsp_id.values:
+        gsp_id = int(gsp_id)
         # make forecast values
         forecast_values = []
 
         # get location
         location = get_location(session=session, gsp_id=gsp_id)
 
-        gsp_forecast_values_df = forecast_values_df.query(f"gsp_id=={gsp_id}")
+        gsp_forecast_values_da = forecast_values_dataarray.sel(gsp_id=gsp_id)
 
-        for i, forecast_value in gsp_forecast_values_df.iterrows():
+        for target_time in pd.to_datetime(gsp_forecast_values_da.target_datetime_utc.values):
             # add timezone
-            target_time = forecast_value.target_datetime_utc.replace(tzinfo=timezone.utc)
+            target_time_utc = target_time.replace(tzinfo=timezone.utc)
+            this_da = gsp_forecast_values_da.sel(target_datetime_utc=target_time)
+
             forecast_value_sql = ForecastValue(
-                target_time=target_time,
-                expected_power_generation_megawatts=forecast_value.forecast_mw,
+                target_time=target_time_utc,
+                expected_power_generation_megawatts=(
+                    this_da.sel(output_label="forecast_mw").item()
+                ),
             ).to_orm()
+
             forecast_value_sql.adjust_mw = 0.0
+
+            forecast_value_sql.properties = {}
+
+            if "forecast_mw_plevel_10" in gsp_forecast_values_da.output_label:
+                forecast_value_sql.properties["10"] = this_da.sel(
+                    output_label="forecast_mw_plevel_10"
+                ).item()
+
+            if "forecast_mw_plevel_90" in gsp_forecast_values_da.output_label:
+                forecast_value_sql.properties["90"] = this_da.sel(
+                    output_label="forecast_mw_plevel_90"
+                ).item()
+
```

Comment:
could you add to the tests, to confirm that "90" is added to the database?

Reply:
Yeh, just added